app/helpers/application_helper.rb

```diff
@@ -24,6 +24,42 @@ def miq_accordion_panel(title, condition, id, &block)
     end
   end
 
+  def single_relationship_link(record, table_name, property_name = nil)
+    out = ''
+    property_name ||= table_name
+    ent = record.send(property_name)
+    name = ui_lookup(:table => table_name.to_s)
+    if role_allows(:feature => "#{table_name}_show") && !ent.nil?
+      out = content_tag(:li) do
+        link_to("#{name}: #{ent.name}",
+                {:controller => table_name, :action => 'show', :id => ent.id.to_s},
+                :title => _("Show this #{record.class.name.demodulize.titleize}'s parent #{name}"))
+      end
+    end
+    out
+  end
+
+  def multiple_relationship_link(record, table_name)
+    out = ''
+    if role_allows(:feature => "#{table_name}_show_list") &&
+       (table_name != 'container_route' || record.respond_to?(:container_routes))
+      plural = ui_lookup(:tables => table_name.to_s)
+      count = record.number_of(table_name.to_s.pluralize)
+      if count == 0
+        out = content_tag(:li, :class => "disabled") do
+          link_to("#{plural} (0)", "#")
+        end
+      else
+        out = content_tag(:li) do
+          link_to("#{plural} (#{count})",
+                  {:action  => 'show', :id => @record, :display => table_name.to_s.pluralize},
+                  :title => _("Show #{plural}"))
+        end
+      end
+    end
+    out
+  end
+
```

Comment:
Hrm, I'm not sure `single_relationship_link` is gonna work correctly ... correct me if I'm missing something.

This block from `app/views/layouts/listnav/_container_group.html.haml` was replaced with a call to `single_relationship_link`:

> ```ruby
>      - if role_allows(:feature => "container_node") && !@record.container_node.nil?
>        %li
>          = link_to("#{ui_lookup(:table => "container_node")}: #{@record.container_node.name}",
>              {:controller => 'container_node', :action => 'show', :id => @record.container_node},
>              :title => _("View %s %s") % [ui_lookup(:table => "container_node"), @record.container_node.name])
> ```

And the corresponding call to `single_relationship_link`:

> ```ruby
>  = single_relationship_link(@record, :container_node)
> ```

Notice that in the original code, the call to `role_allows` does not include `"_show"` in the feature name.  However the implementation of `single_relationship_link` will always append `"_show"` to the feature name.

I think you'll have to include an optional "feature_name" to the arguments for this method, probably.